### PR TITLE
I fixed /64 bug

### PR DESCRIPTION
**Pull request by hidemune**
Thank you for pointing out.

### CODE REVIEW

File: xpad.c [hunk 1]
```diff
@@ -101,7 +102,8 @@
 #define XTYPE_XBOX360     1
 #define XTYPE_XBOX360W    2
 #define XTYPE_XBOXONE     3
-#define XTYPE_UNKNOWN     4
+#define XTYPE_XBOXONEWHL  4
```
Comment (paroj) [inline xpad.c]: do not introduce a new XTYPE_ - if anything this should be a wheel mapping for the XTYPE_XBOXONE

File: xpad.c [hunk 2]
```diff
@@ -231,6 +233,8 @@ static const struct xpad_device {
 	{ 0x24c6, 0x5b03, "Thrustmaster Ferrari 458 Racing Wheel", 0, XTYPE_XBOX360 },
 	{ 0x1532, 0x0a03, "Razer Wildcat for Xbox One", 0, XTYPE_XBOXONE },
 	{ 0xffff, 0xffff, "Chinese-made Xbox Controller", 0, XTYPE_XBOX },
+	{ 0x0738, 0x4503, "Mad Catz Pro Racing Force Feedback Wheel & Pedals (Xbox One)", 0, XTYPE_XBOXONEWHL },
```
Comment (paroj) [inline xpad.c]: probably this line is enough to get the wheel working

File: xpad.c [hunk 3]
```diff
@@ -703,6 +707,61 @@ static void xpadone_process_packet(struct usb_xpad *xpad, u16 cmd, unsigned char
...
+				 !!(data[5] & 0x08) - !!(data[5] & 0x04));
+		input_report_abs(dev, ABS_HAT0Y,
+				 !!(data[5] & 0x02) - !!(data[5] & 0x01));
+
+	/* TL/TR */
+	input_report_key(dev, BTN_TL,	data[5] & 0x10);
+	input_report_key(dev, BTN_TR,	data[5] & 0x20);
+
+	/* Wheel 0000 - *xx7f* - ffff */
+	input_report_abs(dev, ABS_X,
```
Comment (paroj) [inline xpad.c]: why is changing the mapping ABS_Z -> ABS_X necessary?

File: xpad.c [hunk 4]
```diff
@@ -703,6 +707,61 @@ static void xpadone_process_packet(struct usb_xpad *xpad, u16 cmd, unsigned char
...
+			(__s16) (le16_to_cpup((__le16 *)(data + 6))-32768));
+
+	/* Pedal *0000* - FFFF */
+	input_report_abs(dev, ABS_RX,
+			 (__s16) (le16_to_cpup((__le16 *)(data + 8))-32768));
+	input_report_abs(dev, ABS_RY,
```
Comment (paroj) [inline xpad.c]: same as above

**Comment by hidemune** [comment]
On 2017年05月10日 00:01, Pavel Rojtberg wrote:
>
> *@paroj* requested changes on this pull request.
>
> ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
>
> In xpad.c <https://github.com/paroj/xpad/pull/74#discussion_r115511980>:
>
> > @@ -101,7 +102,8 @@
>   #define XTYPE_XBOX360     1
>   #define XTYPE_XBOX360W    2
>   #define XTYPE_XBOXONE     3
> -#define XTYPE_UNKNOWN     4
> +#define XTYPE_XBOXONEWHL  4
>
> do not introduce a new XTYPE_ - if anything this should be a wheel mapping for the XTYPE_XBOXONE
>

I think that the introduction of a new type is indispensable.


> ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
>
> In xpad.c <https://github.com/paroj/xpad/pull/74#discussion_r115512098>:
>
> > @@ -231,6 +233,8 @@ static const struct xpad_device {
>   	{ 0x24c6, 0x5b03, "Thrustmaster Ferrari 458 Racing Wheel", 0, XTYPE_XBOX360 },
>   	{ 0x1532, 0x0a03, "Razer Wildcat for Xbox One", 0, XTYPE_XBOXONE },
>   	{ 0xffff, 0xffff, "Chinese-made Xbox Controller", 0, XTYPE_XBOX },
> +	{ 0x0738, 0x4503, "Mad Catz Pro Racing Force Feedback Wheel & Pedals (Xbox One)", 0, XTYPE_XBOXONEWHL },
>
> probably this line is enough to get the wheel working
>
Yes.

> ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
>
> In xpad.c <https://github.com/paroj/xpad/pull/74#discussion_r115512791>:
>
> > +	input_report_key(dev, BTN_B,	data[4] & 0x20);
> +	input_report_key(dev, BTN_X,	data[4] & 0x40);
> +	input_report_key(dev, BTN_Y,	data[4] & 0x80);
> +
> +	/* digital pad */
> +		input_report_abs(dev, ABS_HAT0X,
> +				 !!(data[5] & 0x08) - !!(data[5] & 0x04));
> +		input_report_abs(dev, ABS_HAT0Y,
> +				 !!(data[5] & 0x02) - !!(data[5] & 0x01));
> +
> +	/* TL/TR */
> +	input_report_key(dev, BTN_TL,	data[5] & 0x10);
> +	input_report_key(dev, BTN_TR,	data[5] & 0x20);
> +
> +	/* Wheel 0000 - *xx7f* - ffff */
> +	input_report_abs(dev, ABS_X,
>
> why is changing the mapping ABS_Z -> ABS_X necessary?
>

this is main Wheel axis.(data + 6)
Left(0000)-Center(ff7f)-Right(ffff)
I want the highest resolution here.
ABS_Z has 0-1023,but I want { -nnn < Center0 < + nnn }


> ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
>
> In xpad.c <https://github.com/paroj/xpad/pull/74#discussion_r115512868>:
>
> > +				 !!(data[5] & 0x08) - !!(data[5] & 0x04));
> +		input_report_abs(dev, ABS_HAT0Y,
> +				 !!(data[5] & 0x02) - !!(data[5] & 0x01));
> +
> +	/* TL/TR */
> +	input_report_key(dev, BTN_TL,	data[5] & 0x10);
> +	input_report_key(dev, BTN_TR,	data[5] & 0x20);
> +
> +	/* Wheel 0000 - *xx7f* - ffff */
> +	input_report_abs(dev, ABS_X,
> +			(__s16) (le16_to_cpup((__le16 *)(data + 6))-32768));
> +
> +	/* Pedal *0000* - FFFF */
> +	input_report_abs(dev, ABS_RX,
> +			 (__s16) (le16_to_cpup((__le16 *)(data + 8))-32768));
> +	input_report_abs(dev, ABS_RY,
>
> same as above
>

Break Pedal(data + 8) and
Accelerator Pedal(data + 10)

Break : ABS_RZ,
The resolution provided by the driver is 65535.

Accelerator: ABS_X, The resolution provided by the driver is 65535.
If we do not introduce a new type, it will be the movement of center, highest value, lowest value, middle.

Any other ideas?

Best Regards.

**Pull request closed without merging**